src/csharp/Grpc.Core.Tests/ChannelConnectivityTest.cs

```diff
@@ -73,6 +73,24 @@ public async Task Channel_WaitForStateChangedAsync()
             Assert.AreEqual(ChannelState.Ready, channel.State);
         }
 
+        [Test]
+        public async Task Channel_TryWaitForStateChangedAsync()
+        {
+            helper.UnaryHandler = new UnaryServerMethod<string, string>((request, context) =>
+            {
+                return Task.FromResult(request);
+            });
+
+            Assert.IsFalse(await channel.TryWaitForStateChangedAsync(channel.State, DateTime.UtcNow.AddMilliseconds(10)));
+
+            var stateChangedTask = channel.TryWaitForStateChangedAsync(channel.State);
+
+            await Calls.AsyncUnaryCall(helper.CreateUnaryCall(), "abc");
```

Comment:
This test seems like it will work fine, but also looks slightly roundabout: e.g. since it relies on some slightly implicit things.

Any thoughts on removing the RPC from this test and instead e.g. calling `ConnectAsync` to transition the state to "Ready".
Further on that note, any thoughts about introducing a `TryConnectAsync` API as well? (it seems a `TryConnectAsync` API would fit in with the rest of this PR).

Reply:
Good point about the test being more complex than needed. I used ConnectAsync to simplify.

About introducing TryConnectAsync: we could add it but no one is actually asking for it and I'd like to prevent increasing the API surface when not necessary. Also one of the use cases for `TryWaitForStateChangedAsync` is continuous monitoring of channel's connectivity state which means that the method will be run in a loop, possibly with short timeout (so not incurring extra overhead when throwing an exception seem important)  I don't see the same problem for ConnectAsync.